src/tarsnapper/script.py

```diff
@@ -50,18 +50,26 @@ def __init__(self, log, options, dryrun=False):
         self._queried_archives = None
         self._known_archives = []
         self.key_passphrase = None
+        self.archive_cache = {}
 
-    def call(self, *arguments):
+    def call(self, *arguments, **kwargs):
         """
         ``arguments`` is a single list of strings.
         """
+        job = kwargs.get('job')
+        if not job:
+            raise TypeError("Please pass job parameter to call()")
         call_with = ['tarsnap']
         for option in self.options:
             key = option[0]
             pre = "-" if len(key) == 1 else "--"
             call_with.append("%s%s" % (pre, key))
             for value in option[1:]:
                 call_with.append(value)
+        if job.keyfile:
+            call_with.extend(['--keyfile', job.keyfile])
+        if job.cachedir:
+            call_with.extend(['--cachedir', job.cachedir])
```

Comment:
@jbzdak Wouldn't it be cleaner to expan these outside of `call()`, so that this method doesn't need to know about jobs?